### DISCUSSION TITLE
Inverted dependencies

### DISCUSSION THREAD

**Discussion by AlexAegis** [vercel/turborepo#5440] [Ideas]
## Goal

I'd like to define global dependencies from inside the packages like this:

> `packages/testrec/turbo.json`

```json
{
    "$schema": "https://turborepo.org/schema.json",
    "extends": [
        "//"
    ],
    "pipeline": {
        "build": {
            "dependedBy": [
                "test"
            ]
        }
    },
}
```

This would result in any `test` task, regardless of the dependency graph, requiring this package to run its `build` script before any `test` task can run.

Now, I'm aware that this can be defined on the top level `turbo.json` file like this:

> `turbo.json`

```json
{
    "$schema": "https://turborepo.org/schema.json",
    "pipeline": {
        "test": {
            "dependsOn": [
                "testrec#build"
            ],
            "outputMode": "new-only"
        }
    }
}
```

## Why?

This however couples this `turbo.json` to this specific project, and having a bunch of projects, I'd like to have a single `turbo.json` that I can copy into all of them.

Some projects have unique dependencies, outside of the dependency tree.

## Workaround

I can define a `turbo.json` file in every package to define this unique dependency

> `packages/*/turbo.json`

```json
{
    "$schema": "https://turborepo.org/schema.json",
    "extends": [
        "//"
    ],
    "pipeline": {
        "test": {
            "dependsOn": [
                "^build",
                "testrec#build"
            ]
        }
    }
}
```

But this means I have to put this in every package, better than nothing, but kinda ugly. And I can forget it when creating a new package.

**Comment by anthonyshew** [comment]
I'm not sure that this is a feature that we would likely pick up. When building the graph, it's usually simplest to model it as unidirectional - and configuring it as such matches that model.

I'll leave this Discussion open to see if this idea picks up steam and will revisit in the future.
